app/controllers/users/backup_code_setup_controller.rb

```diff
@@ -11,12 +11,19 @@ class BackupCodeSetupController < ApplicationController
     before_action :set_backup_code_setup_presenter
     before_action :apply_secure_headers_override
     before_action :authorize_backup_code_disable, only: [:delete]
-    before_action :confirm_recently_authenticated_2fa, except: [:reminder]
 
     helper_method :in_multi_mfa_selection_flow?
+    helper_method :in_account_creation_flow?
 
     def index
```

Comment:
Talking with @kevinsmaster5 on Slack, I think we'll want to revisit this, i.e. not have backup codes regenerated on a GET path. We'll still need to find a workaround for the bug caught at https://github.com/18F/identity-idp/pull/10014#discussion_r1478799722 .

Reply:
+1, I'm catching up with this PR now and seeing a big side effect in a GET/index path definitely jumped out at me

Reply:
@zachmargolis While we wanted to try to avoid the GET request for creating the keys, ultimately we found that the way MFA setup works (initial setup, setting up as a second method in account creation, and reauthentication) involve a lot of redirects such that implementing this as a non-GET method proved to be quite challenging. The compromise we settled on was to implement it as GET, but [check where the user's coming from](https://github.com/18F/identity-idp/pull/10014/files#diff-26cf7fb1c93eefc7da0ac75e24146e7ba9024b45a25da36abdb22e7d97320fe7R80-R84) to make sure they can't accidentally arrive at this page and reset their codes unknowingly.

There was also an [iteration involving an auto-submitting form](https://github.com/18F/identity-idp/pull/10014/commits/982158e5855bfcc1410892d23ac92520157f5eb3#diff-133fec036c429a712530906ad87e841074a3b20ef95baa6a6fdb808f3cd48414) similar to what we have with SAML, but I don't think it really addresses the underlying problems (it still involves a GET that resets codes without user intervention), and caused noticeable content flickering with the interstitial screen.

Curious if you'd have any other ideas or still have concerns about the approach with the safeguards we tried to add..

Reply:
The internal referrer check is probably about the best I can think of, other than user intervention (which I'm guessing might be outside the scope of this PR?)